pkg/utils/node.go

```diff
@@ -47,19 +47,43 @@ func GetNodeObject(clientset kubernetes.Interface, hostnameOverride string) (*ap
 // Order of preference:
 // 1. NodeInternalIP
 // 2. NodeExternalIP (Only set on cloud providers usually)
-func GetNodeIP(node *apiv1.Node) (net.IP, error) {
+func GetNodeIP(node *apiv1.Node, enableIPv4, enableIPv6 bool) (net.IP, net.IP, error) {
```

Comment:
Aren't you changing the algorithm of this function? According to the comment, NodeInternalIP should have priority over NodeExternalIP. However, in your function, if both exist, NodeExternalIP overwrites NodeInternalIP and gets returned

Reply:
If both NodeInternalIP and NodeExternalIP exist, I think there will be no override. Notice the `ipvAddr* == nil` conditions.

In that case:

```go
	if addresses, ok := addressMap[apiv1.NodeInternalIP]; ok {
		for _, address := range addresses {
			ipAddr := net.ParseIP(address.Address)
			// This is true, because ipAddrv4 == nil
			if enableIPv4 && ipAddrv4 == nil && len(ipAddr) == net.IPv4len {
				ipAddrv4 = ipAddr
			}
			// This is true, because ipAddrv6 == nil!
			if enableIPv6 && ipAddrv6 == nil && len(ipAddr) == net.IPv6len {
				ipAddrv6 = ipAddr
			}
		}
	}
	if addresses, ok := addressMap[apiv1.NodeExternalIP]; ok {
		for _, address := range addresses {
			ipAddr := net.ParseIP(address.Address)
			// This is false, because ipAddrv4 is not nil
			if enableIPv4 && ipAddrv4 == nil && len(ipAddr) == net.IPv4len {
				ipAddrv4 = ipAddr
			}
			// This is false, because ipAddrv6 is not nil
			if enableIPv6 && ipAddrv6 == nil && len(ipAddr) == net.IPv6len {
				ipAddrv6 = ipAddr
			}
		}
	}
```

The second loop is going to override any value only if it's `nil`.

Is there something I missed possibly?

Reply:
You are right! Sorry, I confused it